### PR TITLE
Ported t decomp to become a ref

### PR DESCRIPTION
**Pull request by Chillee**
Also added an error input for `t`

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -192,6 +192,7 @@
     "squeeze",
     "tensor_split",
     "transpose",
+    "t",
```
Comment (mruberry) [inline torch/_refs/__init__.py:195]: Nit: "t" then "tensor_split" to preserve alphabetical ordering

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -2038,6 +2039,22 @@ def tensor_split(
...
+    #     dense_dim = a.dense_dim()
+    #     if not (sparse_dim <= 2 and dense_dim == 0):
+    #         raise RuntimeError(
+    #             f"t() expects a tensor with <= 2 sparse and 0 dense dimensions, but got {sparse_dim} sparse and {dense_dim} dense dimensions"
+    #         )
+    if a.ndim > 2:
+        raise RuntimeError(
+            f"t() expects a tensor with <= 2 dimensions, but self is {a.ndim}D"
+        )
+    return torch.transpose(a, 0, 0 if a.ndim < 2 else 1)
```
Comment (mruberry) [inline torch/_refs/__init__.py:2055]: What about 

```
if a.ndim <= 1:
  return prims.view_of(a)
if a.ndim == 2:
  return transpose(a, 0, 1)

raise RuntimeError()
```

?
```

Comment (Chillee) [reply torch/_refs/__init__.py:2055]: Plausibly nicer, but this is how the ATen op is currently implemented: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/TensorShape.cpp#L2503

Comment (Chillee) [reply torch/_refs/__init__.py:2055]: I can add a case to the `transpose` ref to handle when the 2 dims are the same?

Comment (mruberry) [reply torch/_refs/__init__.py:2055]: It's not like we have to implement things exactly as ATen does.

Yeah we can extend the existing short-circuit here instead:

https://github.com/pytorch/pytorch/blob/876c359347faae9367cb390071034e8705bad90e/torch/_refs/__init__.py#L1943

```
if a.ndim <= 1 or dim0 == dim1:
```

File: torch/_refs/__init__.py [hunk 3]
```diff
@@ -2038,10 +2039,27 @@ def tensor_split(
         return tuple(splits)
 
 
+def t(a: TensorLikeType):
+    # Currently refs don't support sparse checks
```
Comment (mruberry) [inline torch/_refs/__init__.py:2043]: I was a little confused about why this comment is here, maybe something like

```
# TODO: enable sparse support 
```

would be clearer?

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -4223,6 +4223,15 @@ def error_inputs_embedding(op_info, device, **kwargs):
         yield ErrorInput(SampleInput(weight, args=(indices,)), error_type=RuntimeError,
                          error_regex="'weight' must be 2-D")
 
+
+def error_inputs_t(op_info, device, **kwargs):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:4227]: Nice error input

**Review by mruberry — APPROVED** [review]
Cool!

File: torch/_refs/__init__.py (hunk 3 above)
Comment (Chillee) [reply torch/_refs/__init__.py:2043]: Sounds good.

**Comment by Chillee** [comment]
@pytorcbot rebase

**Comment by Chillee** [comment]
This doesn't look like a real error to me (master is also failing with errors like this).
```
  test_backward_nn_functional_batch_norm_cuda_float32 (__main__.TestCompositeComplianceCUDA) ... Traceback (most recent call last):
  File "test/run_test.py", line 1079, in <module>
    main()
  File "test/run_test.py", line 1057, in main
    raise RuntimeError(err_message)
RuntimeError: test_ops failed! Received signal: SIGIOT
```

Will merge if no other issues pop up.

**Pull request closed without merging**